youtube_dl/utils.py

```diff
@@ -1835,6 +1835,12 @@ def write_json_file(obj, fn):
                 os.unlink(fn)
             except OSError:
                 pass
+        try:
+            mask = os.umask(0)
+            os.umask(mask)
+            os.chmod(tf.name, 0o666 & ~mask)
+        except OSError:
```

Comment:
Are you sure no other exceptions possible here?

Reply:
I believe that this module will only raise OSError per the docs
https://docs.python.org/3/library/os.html

> Note All functions in this module raise OSError (or subclasses thereof) in the case of invalid or inaccessible file names and paths, or other arguments that have the correct type, but are not accepted by the operating system.